templater/jobs/utils/utils.go

```diff
@@ -48,14 +56,33 @@ func UnmarshalJobs(jobDir string) (map[string]types.JobConfig, error) {
 		fileName := file.Name()
 		filePath := filepath.Join(jobDir, fileName)
 		if strings.Contains(fileName, "1-X") {
-			for i, releaseBranch := range releaseBranches {
+			var data map[string]interface{}
+			var versions []string
+			if strings.Contains(fileName, "golang-1-X") {
```

Comment:
can we separate the `golang-1-X` if statements and the `1-X` statements? 

Meaning, let's have one block that encapsulates all of the logic for `golang-1-X`, and one for `1-X`, rather than having them intermingled with nested if statements (such as the nested check on https://github.com/aws/eks-distro-prow-jobs/pull/419/files#diff-bab6f1e90b878b24feef430b44242f7a3f90d22551f33ab8c831066b7f824333R71) . 

That way we can keep is really clear what pertains to Golang vs other projects, and then we can follow that same pattern when we add other versioned projects